opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -57,13 +58,13 @@ pub trait LogProcessor: Send + Sync + Debug {
     /// # Parameters
     /// - `record`: A mutable reference to `LogData` representing the log record.
     /// - `instrumentation`: The instrumentation library associated with the log record.
-    fn emit(&self, data: &mut LogRecord, instrumentation: &InstrumentationLibrary);
+    async fn emit(&self, data: &mut LogRecord, instrumentation: &InstrumentationLibrary);
```

Comment:
There is already a concern about forcing `async`  everywhere, even in places where it won't do any good, but actually lowers performance. https://github.com/open-telemetry/opentelemetry-rust/issues/2031

Agree that the blocking is an issue that needs to be solved, but I am not convinced that this change is an acceptable solution :(

Reply:
Blocking the runtime in this method is very dicey and will at best have significant negative performance implications and at worst lock things up entirely, and should only be done if no other solutions are present.

Given that this has been observed causing production issues for folks, it's probably best to just take the approach that is correct and optimize it (if benchmarks even suggest that there is a problem) rather than do this trick.

Reply:
Besides, if AsyncDrop does land, it's very easy to just undo this change and make the drop async

Reply:
I am not sure I follow. What is the negative performance implication being referred to here? To be clear, `BatchLogRecordProcessor` is the one that requires a runtime and benefit from async, but that is *not* the only way to export logs. We have exporters to export to Windows ETW, Linux user-events, where exporting is done synchronously, inline within the logging call itself, and has no need for async and runtimes.

Reply:
Agree with @cijothomas. Async runtime was an optional dependency till now - only required for batch*processor (in logs and traces) and periodic-reader (in metrics). And there was plan to have sync and async version of LogExporter::export() method to make it more complete. However, with this change, the runtime would be always required, even for custom synchronous processors.